common/src/main/java/net/william278/huskchat/discord/WebhookDispatcher.java

```diff
@@ -0,0 +1,84 @@
+package net.william278.huskchat.discord;
+
+import net.william278.huskchat.message.ChatMessage;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+
+public class WebhookDispatcher {
+
+    private final Map<String, URL> channelWebhooks;
+
+    private Optional<URL> getChannelWebhook(@NotNull String channelId) {
+        if (channelWebhooks.containsKey(channelId)) {
+            return Optional.of(channelWebhooks.get(channelId));
+        }
+        return Optional.empty();
+    }
+
+    public WebhookDispatcher(@NotNull Map<String, URL> channelWebhooks) {
+        this.channelWebhooks = channelWebhooks;
+    }
+
+    public CompletableFuture<Void> dispatchWebhook(@NotNull ChatMessage message) {
+        return CompletableFuture.runAsync(() -> {
+            getChannelWebhook(message.targetChannelId).ifPresent(webhookUrl -> {
+                System.out.println("dispatching webhook to " + webhookUrl.toString());
```

Comment:
Maybe show this only if there's an error response code (4xx or 5xx), and as it's a bad idea to have secrets in log files, maybe have only the channel ID logged in the event of an error.

Reply:
Yeah, was a debug leftover, cheers.